### PR TITLE
Add protocolSlotLength

### LINKED ISSUE #1345: Take BFT slot length from genesis file rather than hard coding.
The slot length is currently hard coded to 20s:
https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/src/Ouroboros/Consensus/Node.hs#L188

But we can take it from the genesis file, since it is specified there. For example in the mainnet genesis:
```
        "slotDuration": "20000",
```
The unit there is milliseconds.

So this should be a simple matter of plumbing. We already take lots of other values from the genesis file. This one just got missed.

### PR DESCRIPTION
**Pull request by mrBliss**
Fixes #1345.

The `SlotLength` is a parameter of the consensus protocol. For (Real)PBFT, it
is obtained from the genesis config.

### CODE REVIEW

**Comment by mrBliss** [comment]
One of the reasons #953 wasn't merged, is because I was already thinking about how to handle the hard fork, which requires changing the `SlotLength` dynamically. We'll deal with that in #282.

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs [hunk 1]
```diff
@@ -132,6 +133,11 @@ protocolInfoByron genesisConfig@Genesis.Config {
                                        $ genesisKeyHashes
               , pbftSignatureThreshold = unSignatureThreshold $
                   fromMaybe defaultPBftSignatureThreshold mSigThresh
+              , pbftSlotLength         = slotLengthFromMillisec
+                                       . fromIntegral
```
Comment (dcoutts) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs]: These conversions can be subtle, but a nice trick to make it clear to the reader is to use an explicit type annotation:
```
  . (fromIntegral :: Natural -> Integer)
```
Then I can see there's no problem here without having to look up what `Update.ppSlotDuration` gives us.

File: ouroboros-consensus/test-consensus/Test/Dynamic/LeaderSchedule.hs [hunk 2]
```diff
@@ -40,6 +40,7 @@ tests = testGroup "Dynamic chain generation"
       , praosSlotsPerEpoch = 3
       , praosLeaderF       = 0.5
       , praosLifetimeKES   = 1000000
+      , praosSlotLength    = slotLengthFromSec 2
```
Comment (dcoutts) [inline ouroboros-consensus/test-consensus/Test/Dynamic/LeaderSchedule.hs]: Was this supposed to be 2 or 20? Just checking.

File: ouroboros-consensus/test-consensus/Test/Dynamic/Praos.hs [hunk 3]
```diff
@@ -80,6 +80,7 @@ tests = testGroup "Dynamic chain generation"
       , praosSlotsPerEpoch = 3
       , praosLeaderF       = 0.5
       , praosLifetimeKES   = 1000000
+      , praosSlotLength    = slotLengthFromSec 2
```
Comment (dcoutts) [inline ouroboros-consensus/test-consensus/Test/Dynamic/Praos.hs]: Same double-checking question here.

File: ouroboros-consensus/test-consensus/Test/Dynamic/LeaderSchedule.hs (hunk 2 above)
Comment (mrBliss) [reply ouroboros-consensus/test-consensus/Test/Dynamic/LeaderSchedule.hs]: 2, as the slot length of Praos will also be 2s, right?

File: ouroboros-consensus/test-consensus/Test/Dynamic/Praos.hs (hunk 3 above)
Comment (mrBliss) [reply ouroboros-consensus/test-consensus/Test/Dynamic/Praos.hs]: Same answer

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs (hunk 1 above)
Comment (mrBliss) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs]: I have added the annotation.

**Pull request merged**